File: backend-spring-boot/src/main/java/com/facundocastro/api/model/Tecnologia.java
package com.facundocastro.api.model;

import com.sun.istack.NotNull;
import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Data
@Entity
public class Tecnologia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private String nombre;
    private String color;
    @NotNull
    private String clase;
}

File: frontend-angular/src/app/models/Tecnologia.ts
export class Tecnologia {
  protected _id: number = -1;
  protected _nombre: string = '';
  protected _color: string = '';
  protected _clase: string = '';

  public constructor() {}

  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }

  public get nombre(): string {
    return this._nombre;
  }
  public set nombre(value: string) {
    this._nombre = value;
  }

  public get color(): string {
    return this._color;
  }
  public set color(value: string) {
    this._color = value;
  }

  public get clase(): string {
    return this._clase;
  }
  public set clase(value: string) {
    this._clase = value;
  }
}
